[dynamo] Remove workaround for `functools.wraps` in functorch

StrongerXi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142014
* #142000

This is no longer needed after #142000.

Fixes #123365.

D66838774

StrongerXi committed:
Update

[ghstack-poisoned]

StrongerXi commented on torch/_functorch/eager_transforms.py:
```diff
@@ -761,12 +761,7 @@ def compute_jacobian_preallocate_and_copy():
-    if not torch._dynamo.is_compiling():
-        wrapper_fn = wraps(func)(wrapper_fn)
-
+    wrapper_fn = wraps(func)(wrapper_fn)
```
Ah right, much cleaner, updated.

StrongerXi committed:
Update

[ghstack-poisoned]

StrongerXi commented:
@pytorchbot merge

StrongerXi commented:
@pytorchbot merge

atalman commented:
@pytorchmergebot -c ghfirst -m "Sorry https://github.com/pytorch/pytorch/pull/142000 is failing internally, need to revert this"

atalman commented:
@pytorchmergebot revert -c ghfirst -m "Sorry https://github.com/pytorch/pytorch/pull/142000 is failing internally, need to revert this"

StrongerXi commented:
@StrongerXi has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D66838610).

StrongerXi committed:
Update on "[dynamo] Remove workaround for `functools.wraps` in functorch"


This is no longer needed after #142000.

Fixes #123365.

Differential Revision: [D66838774](https://our.internmc.facebook.com/intern/diff/D66838774)

[ghstack-poisoned]

izaitsevfb commented:
@pytorchbot merge -f "unrelated failures"

izaitsevfb commented:
@pytorchbot merge -f "unrelated failures"

let's try one more time

izaitsevfb commented:
@pytorchbot merge -f "unrelated failures"

third time's a charm 
(removed "Differential revision" from the second PR in the stack)
